test/AutoRest.TestServerLowLevel.Tests/dpg-customization.cs

```diff
@@ -24,10 +24,34 @@ public class DpgCustomizationTest : TestServerLowLevelTestBase
         [Test]
         public Task GetHandwrittenModel() => Test(async (host) =>
         {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            var scopes = diagnosticListener.Scopes;
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
             Response<Product> result = await new DPGClient(Key, host).GetModelValueAsync("model");
+            Assert.AreEqual(1, scopes.Count);
```

Comment:
Hi @ShivangiReja Why can we according to the scopes.Count to validate if the scope is suppressed?  I could not find any code in Azure.Core that diagnosticListener will not add the scope which is suppressed.

Reply:
`diagnosticListener` is not used to suppress the scope, it is used to listen how many scopes are created, count of the scopes and the scopes are completed or not. We are using it to test the scopes.  The code for ClientDiagnosticListener is in autorest.csharp repo [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/TestFramework/ClientDiagnosticListener.cs). And, the logic to suppress the scope is in the [DiagnosticScope](https://github.com/Azure/azure-sdk-for-net/pull/28066/files#diff-2917d0ddf4987f75fc3ef2ac6445df3150798ecdf01edb8c1a24b2e64cbc72de)

Inside `GetModelValueAsync` method, we are creating two scopes but we are suppressing the nested scopes [here](https://github.com/Azure/autorest.csharp/blob/308f082d8f98f707a11f781021068b9524f57c97/test/TestServerProjectsLowLevel/dpg-customization/Generated/DPGClientOptions.cs#L30). So, in [this](https://github.com/Azure/autorest.csharp/blob/308f082d8f98f707a11f781021068b9524f57c97/test/AutoRest.TestServerLowLevel.Tests/dpg-customization.cs#L32-L34) test case we are validating the scope count is one, the scope name and whether or not it is completed using `diagnosticListener`.